C++/C++.sublime-syntax

```diff
@@ -1044,6 +1050,12 @@ contexts:
     - match: '{{identifier}}(?={{data_structures_forward_decl_lookahead}})'
       scope: entity.name.struct.forward-decl.c++
       set: data-structures-struct-definition-after-identifier
+    - match: '(?={{path_lookahead}}\s*::\s*{{identifier}})'
+      set:
+        - meta_scope: entity.name.struct.c++
+        - include: identifiers
+        - match: '(?=[^\w\s])'
+          set: data-structures-struct-definition-after-identifier
```

Comment:
I tested this out locally and it seems to work fine. Nice job! Only thing missing IMO are unions:

```
namespace ns {
class Bar;
struct Foo;
union Qux;
} // namespace ns

class ns::Bar {
public:
  Bar() = default;
};

struct ns::Foo {
  Foo() = default;
};

union ns::Qux {
  Qux() = default;
};
```

Reply:
Okay the latest commit added support for both enums and unions. Should be good to go now.